client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -23,6 +30,112 @@ import './style.scss';
 
 const debug = debugFactory( 'calypso:checkout-sidebar-plan-upsell' );
 
+const PromoCardV2 = styled.div`
+	position: relative;
+	background-color: #bbe0fa;
+	border-radius: 4px;
+	font-size: 12px;
+	font-weight: 500;
+	align-self: flex-end;
+
+	p {
+		margin: 0;
+		padding: 8px 16px;
+	}
+
+	&:before {
+		background-color: #bbe0fa;
+		display: block;
+		top: -5px;
+		content: '';
+		height: 8px;
+		right: 8%;
+		margin-left: -4px;
+		position: absolute;
+		transform: rotate( 225deg );
+		width: 8px;
+	}
+`;
+
+const CheckoutPromoCard: React.FC< {
+	onUpgradeClick: () => void;
+	responseCart: ResponseCart;
+	currentVariant: WPCOMProductVariant;
+	percentSavings: number;
+} > = ( { onUpgradeClick, responseCart, currentVariant, percentSavings } ) => {
+	const translate = useTranslate();
+	const plan = responseCart.products.find(
+		( product ) => isPlan( product ) && ! isJetpackPlan( product )
+	);
+
+	const domainRegistrationOrTransferInCart = responseCart.products.find(
+		( product ) => isDomainRegistration( product ) || isDomainTransfer( product )
+	);
+
+	const isMonthly = currentVariant?.termIntervalInMonths === 1;
+	const isYearly = currentVariant?.termIntervalInMonths === 12;
+	const isBiennially = currentVariant?.termIntervalInMonths === 24;
+	const isTriennially = currentVariant?.termIntervalInMonths === 36;
+
+	let labelText;
+
+	// TODO: this currently doesn't work because the !plan guard found in CheckoutSidebarPlanUpsell disallows this to be reached.
+	if ( ! plan && domainRegistrationOrTransferInCart ) {
+		labelText = translate(
+			'Longer plan billing cycles save you money and include a custom domain for free for the first year.'
+		);
+	}
+
+	if ( isMonthly ) {
+		labelText = translate(
+			'Longer plan billing cycles save you money and include a custom domain for free for the first year.'
+		);
+
+		if ( domainRegistrationOrTransferInCart ) {
+			labelText = translate(
+				'Save money with longer billing cycles and get %(domainMeta)s free for the first year by {{upgradeLink}}switching to an annual plan.{{/upgradeLink}}',
+				{
+					comment: '"domainMeta" is the domain name and TLD, like "example.com" or "example.org"',
+					args: {
+						domainMeta: domainRegistrationOrTransferInCart.meta,
+					},
+					components: {
+						upgradeLink: <Button onClick={ onUpgradeClick } variant="link" />,
+					},
+				}
+			);
+		}
+	}
+
+	if ( isYearly ) {
+		labelText = translate(
+			"Save up to %(savingsPercentage)s% on longer billing cycles! It's hassle-free and easy on your wallet. {{upgradeLink}}Switch to a two-year plan and save.{{/upgradeLink}}",
+			{
+				comment:
+					'"savingsPercentage is the savings percentage for the upgrade as a number, like "20" for 20%"',
+				args: {
+					savingsPercentage: percentSavings,
+				},
+				components: {
+					upgradeLink: <Button onClick={ onUpgradeClick } variant="link" />,
+				},
+			}
+		);
+	}
+
+	if ( isBiennially || isTriennially ) {
```

Comment:
Not sure if this is really needed, but I included it to ensure we return without showing the promo if we have no label to show. I could probably just lose the condition and cover all non monthly/annual use cases.